src/sources/http.rs

```diff
@@ -61,8 +62,19 @@ impl HttpSource for SimpleHttpSource {
 }
 
 #[typetag::serde(name = "http")]
+#[async_trait]
 impl SourceConfig for SimpleHttpConfig {
     fn build(
+        &self,
+        _name: &str,
+        _globals: &GlobalOptions,
+        _shutdown: ShutdownSignal,
+        _out: mpsc::Sender<Event>,
+    ) -> crate::Result<super::Source> {
+        unimplemented!()
+    }
+
+    async fn build_async(
```

Comment:
This is not for this PR

Reply:
Why? `HttpSource::run` is async now:
https://github.com/timberio/vector/blob/fb97d37dc30a700bd74eda9bd37f42faa2cd654e/src/sources/util/http.rs#L57

Reply:
Oh, my bad. I misread this and the other two as a remnant of the other PR that you had maybe only partially reverted.

Reply:
One note about this change that now `bind` happened in `build_async`, I remember that `future::lazy` was used to remove such effect before -- is this change ok?